doc/source/cd/background.md

```diff
@@ -0,0 +1,862 @@
+demonstrated below, where the offline [MMD
+detector](methods/mmddrift.ipynb) is used to monitor drift in data
+sampled from a normal distribution
+$\mathcal{N}\left(\mu,\sigma^2 \right)$, with the mean starting to
+drift from $\mu=0$ to $\mu=0.5$ at $t=40$.
+
+```{image} images/offline_drift_W2.*
+:align: center
+:alt: Offline detector with W=2
+:width: 700px
+```
+
+```{image} images/offline_drift_W20.*
+:align: center
+:alt: Offline detector with W=20
+:width: 700px
+```
+
+### Online drift detectors
+
+An alternative strategy is to perform a test each time data arrives.
+However the usual offline methods are not applicable because the process
+for computing $p$-values is too expensive. Additionally, they
+don’t account for correlated test outcomes when using overlapping
+windows of test data, leading to miscalibrated detectors operating at an
+unknown False Positive Rate (FPR). Well-calibrated FPR’s are crucial for
+judging the severity of a drift detection. In the absence of
+calibration, drift detection can be useless since there is no way of
+knowing what fraction of detections are false positives. [Alibi
+Detect](https://github.com/SeldonIO/alibi-detect) offers specialist
+online drift detectors, which are designed to be well-calibrated when
+used sequentially:
+
+- [Online Maximum Mean Discrepancy](methods/onlinemmddrift.ipynb)
+- [Online Least-Squares Density
+  Difference](methods/onlinelsdddrift.ipynb)
+
+Thoughtful structuring of the computations reduces the cost of these
+detectors from $O(N^2B)$ to $O(N^2 + NB)$ during
+configuration (where $B$ is the number of bootstrap samples used
+for calibration), and from $O(N^2)$ to $O(N)$ during
+operation. The detectors compute a test statistic $S(\mathbf{x})$
+during the configuration phase. Then, at test time, the test statistic
+is updated sequentially at a low cost. When no drift has occurred the
+test statistic fluctuates around its expected value, and once drift
+occurs the test statistic starts to drift upwards. When it exceeds some
+preconfigured threshold value, drift is detected. The online detectors
+are constructed in a similar manner to the offline detectors, for
+example for the online MMD detector:
+
+```python
+online_detector = MMDDriftOnline(X_ref, ert, window_size, backend='tensorflow', n_bootstraps=5000)
+```
+
+But, in addition to providing the detector with reference data, the
+expected run-time (see below), and size of the sliding window must also
+be defined. Another important difference is that the online detectors
+make predictions on single data instances:
+
+```python
+result = online_detector.predict(X[i])
+```
+
+This can be seen in the animation below, where the online detector
+considers each incoming observation/sample individually, instead of
+considering a batch of observations like the offline detectors.
+
+```{image} images/online_drift.*
+:align: center
+:alt: Online detector
+:width: 700px
+```
```

Comment:
Cool animation but I don't understand what n in the bottom left representing? Also I don't know how easy it would be but would be cool if the translucent window followed behind the line to showing how it slides continuously in contrast to the jumping ones above.